app/code/Magento/Review/Observer/CatalogBlockProductListCollectionInitializeObserver.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Review\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+
+class CatalogBlockProductListCollectionInitializeObserver implements ObserverInterface
+{
+    /**
+     * Store manager
+     *
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        \Magento\Store\Model\StoreManagerInterface $storeManager
+    ) {
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * Join review summary to collection
+     *
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return $this
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $productCollection = $observer->getEvent()->getCollection();
+        if ($productCollection instanceof \Magento\Catalog\Model\ResourceModel\Product\Collection && !$productCollection->isLoaded()) {
```

Comment:
What happens if collection is loaded? How do we add review data then?

Reply:
If collection is loaded than `ReviewRenderer.php` load review for each product in https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Review/Block/Product/ReviewRenderer.php#L77